factory.py

```diff
@@ -64,14 +83,22 @@ def __pre_init__(self, *args, **kwargs):
         """Pre-initialization method. Use/extend this method to perform any checks before initializing the class and
         fail fast to adapt to other factory classes."""
         kw = args[0]
-        assert (
-            FactoryTypes(kw.get(FactoryArgumentKeys.TASK, None)) == self._type
-        ), "Invalid task type."
 
-        algorithm_name = kw.get(FactoryArgumentKeys.ALGORITHM, None)
-        assert (
-            self.get_algorithm(algorithm_name) is not None
-        ), f"Invalid algorithm: {algorithm_name}"
+        if kw.get(FactoryArgumentKeys.JOB_SERVICE, None) == JobServices.ADAPT.value:
```

Comment:
uhh? 🤔 I don't get the issue with this one

Reply:
inconsistent. we do access the .value when we need the string, here if the enum is created well, we can do a direct enum-enum match rather than its value to string match.

Reply:
```
if kw.get(FactoryArgumentKeys.JOB_SERVICE, None) == JobServices.ADAPT.value:
            assert (
                FactoryTypes(kw.get(FactoryArgumentKeys.JOB_SERVICE, None))
                == self._type
            ), "Invalid task type."

        else:
            assert (
                FactoryTypes(kw.get(FactoryArgumentKeys.TASK, None)) == self._type
            ), "Invalid task type."

            algorithm_name = kw.get(FactoryArgumentKeys.ALGORITHM, None)
            assert (
                self.get_algorithm(algorithm_name) is not None
            ), f"Invalid algorithm: {algorithm_name}"
```

Inconsistent how ? This is the exact asmaae way this was being written earlier too

Reply:
ref - https://github.com/nyunAI/nyuntam/blob/32a06b0218de4c3e3fc354f680dd72fb28e61ded/factory.py#L67

we compared the FactoryType(task) with _type which is also a FactoryType

Reply:
done @shwu-nyunai